src/helper.js

```diff
@@ -314,8 +315,19 @@ function parseHostmask(hostmask) {
 
 function compareHostmask(a, b) {
 	return (
-		(a.nick.toLowerCase() === b.nick.toLowerCase() || a.nick === "*") &&
-		(a.ident.toLowerCase() === b.ident.toLowerCase() || a.ident === "*") &&
-		(a.hostname.toLowerCase() === b.hostname.toLowerCase() || a.hostname === "*")
+		compareWithWildcard(a.nick, b.nick) &&
+		compareWithWildcard(a.ident, b.ident) &&
+		compareWithWildcard(a.hostname, b.hostname)
 	);
 }
+
+function compareWithWildcard(a, b) {
+	a = a.toLowerCase();
+	b = b.toLowerCase();
+	const wildcars_split = a.split("*").map((sub) => {
+		const parts = sub.split("?").map((p) => _.escapeRegExp(p));
+		return parts.join(".");
+	});
+	const re = new RegExp(wildcars_split.join(".*"));
```

Comment:
If you split out the part of this function that generates the regex into its own function, you could write tests for that function that would be a bit easier to verify the correctness of.

If you pass `"i"` as a flag to `RegExp` it'll do a case insensitive comparison. Since we're using the regex engine now anyways we might as well let it handle that part of it.

Reply:
ack, will do. Thanks

Reply:
I didn't split the function but added the regex modifier (and a huge comment).
Considering that there are already extensive tests for the comparison function with quite a bunch of edge case scenarios I don't see a benefit of testing for the raw regex, we already know that we get the desired result. Hope that's ok.